manifests/pulp.pp

```diff
@@ -129,6 +142,52 @@
     }
   }
 
+  if $expose_mongodb {
+    include certs::mongodb
+    include certs::mongodb_client
+
+    Class['certs::mongodb_client'] ~> Class['pulp', 'apache::service']
+
+    $mongodb_bind_ip = $expose_mongodb_ipv6 ? {
+      true  => ['0.0.0.0', '::0'],
+      false => ['0.0.0.0'],
+    }
+
+    class { 'mongodb::server':
+      bind_ip => $mongodb_bind_ip,
+      ipv6    => $expose_mongodb_ipv6,
+      ssl     => true,
+      ssl_key => $certs::mongodb::mongodb_server_bundle,
+      ssl_ca  => $certs::mongodb::mongodb_server_ca_cert,
+      require => Class['certs::mongodb'],
+    }
+
+    $mongodb_seeds_real = "${certs::mongodb::hostname}:27017"
+    $db_password = extlib::cache_data('foreman_cache_data', 'mongo_db_password', extlib::random_password(32))
```

Comment:
We already have `$mongodb_password` as a parameter and IMH we should at least prefer that if specified.

Reply:
That password is intended for the use case when a user is providing credentials for an external database. I don't think it should apply to this workflow.

Reply:
I think it does. You're exposing the user with a certain password. The parameter allows the end user to specify something. That makes it easier on the consuming node to use the same password. Now the end user needs to look in files to see which password was generated.

More concrete: we have `server-1.example.com` which runs Katello with Pulp and MongoDB. The user runs `foreman-installer --katello-pulp-mongodb-password mysecretpassword`.

Now the user wants to consume this on `server-2.example.com`. The user runs `foreman-installer --foreman-proxy-content-mongo-db-password mysecretpassword`.

The current implementation doesn't support this because it's silently ignored.

Reply:
My point, unless I am just missing something is, that nobody should be doing this other than our own tooling which is tightly controlled for a very specific upgrade scenario.

Reply:
So how do you get the password now? Do you parse it out of a config file later on?

Reply:
That will be the most likely tactic given it's the only reliable way to get the value. I think I get what you are saying now but not form the users perspective, but the *if* the parameter is used then it can be stored on the answers file. But I don't see how you set the parameter conditionally.